src/components/Charts/ProposalVPChart.tsx

```diff
@@ -0,0 +1,183 @@
+import React, { useEffect, useMemo, useRef, useState } from 'react'
+import { Chart } from 'react-chartjs-2'
+
+import { ChartData, Chart as ChartJS, ScriptableTooltipContext } from 'chart.js'
+import 'chart.js/auto'
+import 'chartjs-adapter-dayjs-4/dist/chartjs-adapter-dayjs-4.esm'
+import annotationPlugin from 'chartjs-plugin-annotation'
+
+import { Vote } from '../../entities/Votes/types'
+import useAbbreviatedFormatter from '../../hooks/useAbbreviatedFormatter'
+import useFormatMessage from '../../hooks/useFormatMessage'
+import useProfiles from '../../hooks/useProfiles'
+import { Avatar } from '../../utils/Catalyst/types'
+import Section from '../Proposal/View/Section'
+
+import './ProposalVPChart.css'
+import {
+  HOUR_IN_MS,
+  externalTooltipHandler,
+  getAbstainColor,
+  getDataset,
+  getNoColor,
+  getSegregatedVotes,
+  getSortedVotes,
+  getYesColor,
+} from './ProposalVPChart.utils'
+
+ChartJS.register(annotationPlugin)
+
+interface Props {
+  requiredToPass?: number | null
+  voteMap: Record<string, Vote>
+  isLoadingVotes?: boolean
+  startTimestamp?: number
+  endTimestamp?: number
+}
+
+const COMMON_DATASET_OPTIONS = {
+  fill: false,
+  stepped: 'before' as const,
+  pointHoverRadius: 3,
+  pointHoverBorderWidth: 8,
+}
+
+function ProposalVPChart({ requiredToPass, voteMap, isLoadingVotes, startTimestamp, endTimestamp }: Props) {
+  const t = useFormatMessage()
+  const YAxisFormat = useAbbreviatedFormatter()
+  const chartRef = useRef<ChartJS>(null)
+  const votes = useMemo(() => getSortedVotes(voteMap), [voteMap])
+  const { profiles, isLoadingProfiles } = useProfiles(votes.map((vote) => vote.address))
+  const profileByAddress = useMemo(
+    () =>
+      profiles.reduce((acc, { profile }) => {
+        acc.set(profile.ethAddress.toLowerCase(), profile)
+        return acc
+      }, new Map<string, Avatar>()),
+    [profiles]
+  )
+  const { yesVotes, noVotes, abstainVotes } = useMemo(
+    () => getSegregatedVotes(votes, profileByAddress),
+    [profileByAddress, votes]
+  )
+  const segregatedVotesMap = {
```

Comment:
getSegregatedVotes could return this object.
`const votes = getSegregatedVotes(...) `
And use it like `getDataset(votes.yes, endTimestamp)`